### ISSUE TITLE
cancel tickets: strange reaction!

### ISSUE THREAD

**Issue by jojo12** [jemproject/JEM-Project#1593]
![grafik](https://github.com/jemproject/JEM-Project/assets/2960236/7f7c4a3a-e2b3-4e2a-8fb6-ed3d32115580)
I have booked 3 places. Now I try to cancel.
the error is: the add places button shows me a modal window and refuses my canceling!

**Comment by jojo12** [comment]
needs more testing! unlogic comportment!

**Comment by jojo12** [comment]
fix above (2469a38) is only in legacy, there it's fixed.

next issue: (when min/user=1 max/user=1 Problem: WAITINGLIST
if waitinglist=yes --> when attending after canceling, then user is only on waitinglist, but should be on attendinglist.
if waitinglist=no  --> when attending after canceling, then user is on attendinglist! This is ok!

(when min/user=1 max/user=2)
if waitinglist=yes --> when attending, but booking additional place, then user and new place is on waitinglist, but should be on    attendinglist
if waitinglist=no --> when attending, but booking additional place, then user and new place is on attendinglist. This is ok

**(waitinglist should only be used, if there's not enough place on attendinglist /or if admin decides to put a user on waitinglist)**

**Comment by jojo12** [comment]
next issue: the button "Register" should be renamed, because we use it for registering and unregistering for events.
my proposal "Send"

**Comment by jojo12** [comment]
next issue: in editevent I invite a user. It's possible to invite him with 0 or 1 place. But 0 place doesn't make sense!

**Comment by mckillo** [comment]
Fixed in default too.

I have to study the waiting list (simulate case)... let's me test it.

**Comment by jojo12** [comment]
waitinglist seems to be the problem only when max.user=0

**Comment by jojo12** [comment]
fixed in reponsive too (32fe7f5)

**Comment by mckillo** [comment]
Ok, I had not refreshed the issue page, and I just saw that it is when max.user=0. Let's me test it again.

**Comment by mckillo** [comment]
But, It hasn't sense to have a waiting list when the Max.Places is 0. Waiting list value should be ignored when Max. places is 0.

**Comment by mckillo** [comment]
If Max.places is 0, waiting lislt should be disable or save as disable.

**Comment by jojo12** [comment]
It's after the last step you made. When you try to attend again! then you're only on the waitinglist!

**Comment by mckillo** [comment]
But any user should be on a waiting list if Max. Places is 0.
If it succeeds because the admin have changed the registration setting of the event, right?

**Comment by jojo12** [comment]
No: we said: max. places is: there's no limit of participants!

**Comment by mckillo** [comment]
Yes, if max.places is 0, the number of places for event is unlimited and for this reason, waiting list should be empty.

![JEM4 - There is a waiting list because max_places is defined is greater than 0 and the booked places is same number](https://github.com/jemproject/JEM-Project/assets/2765316/7cf2f520-80de-415f-8396-2dee1bd32891)

**Comment by mckillo** [comment]
I think that I understand you. Now, if the admin change to Max.places =0 in the event, the user C doesn't change to booked places and only can cancel, but it's normal.

When the user registers a places, the status of this places doesn't change.

The admin should change all users in the waiting list to attendees (manual by admin or wait to next registration, the plugin should change it with the new status).

If user with booked places on the waiting list wants to change to booked places, first the user has to cancel the waiting places and after that he can book the max. booked places.

**Comment by mckillo** [comment]
I think that I simulate the case....

**Comment by jojo12** [comment]
![grafik](https://github.com/jemproject/JEM-Project/assets/2960236/4da54ded-1a49-46e2-8167-5e176dd965c9)

**Comment by jojo12** [comment]
the settings were not changed:
![Uploading grafik.png…]()

**Comment by mckillo** [comment]
@jojo12 Fixed. Check it.

**Comment by jojo12** [comment]
error: Unmatched ')'

**Comment by mckillo** [comment]
Sorry, Ctrl+V didn't work.... Solved. See las commit above

**Comment by jojo12** [comment]
yes, it works, thanks
I continue to test!
